src/main/java/com/google/jenkins/plugins/computeengine/InstanceConfiguration.java

```diff
@@ -148,11 +150,13 @@
   public Integer retentionTimeMinutes;
   public Integer launchTimeoutSeconds;
   public Long bootDiskSizeGb;
+  private String id;
   public transient Set<LabelAtom> labelSet;
   protected transient ComputeEngineCloud cloud;
 
   @DataBoundConstructor
   public InstanceConfiguration(
+      String id,
```

Comment:
I would say that this is not required for the constructor, and should only be set through the DataboundSetter, because it's not something that the user can provide and won't exist until the InstanceConfiguration is created. There should also probably be a hidden field in the UI with this included. It's a bit of a unique situation in the sense that it's generated and not configured by the user.

Reply:
The only thing I'm unsure of is whether:
1. The set... method is called with null for newly submitted forms that don't have the value set.
2. The set... method is not called at all when not provided through the UI.
I think that including a hidden UI field might mitigate that uncertainty though.

Reply:
You can try setting it as in #51

Reply:
If you have setter im not sure if its needed in constructor